core/chaincode/shim/chaincode.go

```diff
@@ -572,11 +572,11 @@ func (stub *ChaincodeStub) GetRows(tableName string, key []Column) (<-chan Row,
 	if err != nil {
 		return nil, fmt.Errorf("Error fetching rows: %s", err)
 	}
-	defer iter.Close()
 
 	rows := make(chan Row)
 
 	go func() {
+		defer iter.Close()
 		for iter.HasNext() {
 			_, rowBytes, err := iter.Next()
 			if err != nil {
```

Comment:
I just want to comment that this PR did not work for us. The problem is that we query two tables consequtively, and the second query fails because (IMU) the access to the first table has not finished.